clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/PartitionConsumptionState.java

```diff
@@ -833,4 +837,18 @@ public void setLastLeaderCompleteStateUpdateInMs(long lastLeaderCompleteStateUpd
   public String getReplicaId() {
     return replicaId;
   }
+
+  public void addIncPushVersionToPendingReportList(String incPushVersion) {
```

Comment:
Shall we do the filtering on the fly? It means we would always keep at most 50 most-recent inc push versions.
With this way, the memory usage will be bounded even with the very extreme situation, such as 1M inc push versions before reporting completed and it will always reduce the size of `OffsetRecord`.
This comment applies to DaVinci as well.